hal/lib/athena/AnalogInternal.cpp

```diff
@@ -23,18 +25,43 @@ IndexedHandleResource<HAL_AnalogInputHandle, hal::AnalogPort, kNumAnalogInputs,
 
 static int32_t analogNumChannelsToActivate = 0;
 
-bool analogSystemInitialized = false;
+static std::atomic<bool> analogSystemInitialized{false};
+
+bool analogSampleRateSet = false;
 
 /**
  * Initialize the analog System.
  */
 void initializeAnalog(int32_t* status) {
+  if (analogSystemInitialized) return;
   std::lock_guard<priority_recursive_mutex> sync(analogRegisterWindowMutex);
   if (analogSystemInitialized) return;
   analogInputSystem.reset(tAI::create(status));
   analogOutputSystem.reset(tAO::create(status));
   setAnalogNumChannelsToActivate(kNumAnalogInputs);
-  HAL_SetAnalogSampleRate(kDefaultSampleRate, status);
+  // Manually set analog sample rate in order to avoid recursion loop
```

Comment:
This feels complex enough that it would likely be better to have an internal function that both HAL_SetAnalogSampleRate and this function call.